[hDJkPkOQ] Add QueryType validation in apoc.trigger.add

vga91 opened the pull request:
Added QueryType validations in validateQuery() method,
so as to fail if the procedure tries to execute schema operations.

See point 2 here: https://github.com/neo4j/apoc/pull/208#pullrequestreview-1145222227

Only for 4.4, since the `apoc.trigger.add` is deprecated and in 5.x the `validateQuery()` is not longer present.

AzuObs commented on core/src/test/java/apoc/trigger/TriggerNewProceduresTest.java:
```diff
@@ -583,4 +583,18 @@ public void testEventualConsistency() {
+                    (row) -> fail("Should fail because of unsupported schema operation"));
+        } catch (RuntimeException e) {
+            final String expected = "Failed to invoke procedure `apoc.trigger.install`: " +
+                    "Caused by: java.lang.RuntimeException: Supported query types for the operation are [READ_ONLY, WRITE, READ_WRITE]";
```
Is there anyway that this could be a more user friendly error message? 

`"The trigger statement must contain READ_ONLY, WRITE, or READ_WRITE query."`

vga91 commented:
> Is there anyway that this could be a more user friendly error message?
> `"The trigger statement must contain READ_ONLY, WRITE, or READ_WRITE query."`

1) while I was making this change,
I realized that it would be better to add this thing to the validate too (https://github.com/neo4j/apoc/commit/e5264e06a1423ebf9f9f8eef88e020e84fa83d28),
i.e. I added `Set.of(Mode.WRITE, Mode.READ, Mode.DEFAULT),` and `testTriggerInstallWithSchemaProcedure` .
Basically as it was before it didn't give an error with a `@Procedure(Mode=SCHEMA)` e.g. `apoc.schema.assert`

2) Along with that, I've added two parameters to the `Util.validateQuery()` with custom error messages.

If you think point `1.` is unnecessary / to be separated / another card needs to be created, I'll be happy to change the pr :)

AzuObs reviewed (approved):
Thank you for the change! Looks good 🙂
